pkgs/development/python-modules/segyio/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ninja
+, python
+, scikit-build
+, pytest
+, numpy
+}:
+
+stdenv.mkDerivation rec {
```

Comment:
Why are you not using `buildPythonPackage` or `buildPythonApplication` instead :thinking: ?

Reply:
Because the package uses cmake instead of having a setup.py.

Reply:
I'm saying this because I have previously tried to use buildPythonPackage, but it assumes there is a setup.py, which in this case is called by the cmake builder and it is not in the root of the repo. I had to explicitly write the buildPhase and installPhase and it still didn't work, using the stdenv and using the toPython function solved the problem.

This is similar to what is done in the [pyside2](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/development/python-modules/pyside2/default.nix#L36) derivation.

Reply:
Huh... Ok, this makes sense.

So I am not comfortable to review this, because I am not sure how `stdenv.mkDerivation` interacts with all the matrix of Pythons we need to build (3.8, 3.9, 3.10, etc). I am assuming this works since `pyside2` uses it, but I prefer to have a reviewer that knows better than me how Python packaging works in Nixpkgs.

CC @SuperSandro2000.

Reply:
Thx for your review as always, let us wait for @SuperSandro2000 to save us all.

Reply:
There is a setup.py https://github.com/equinor/segyio/blob/1.9.9/python/setup.py. I think we want to use that dir as source. Also most likely ``dontUseCmakeConfigure = true;`` is required to run the setup.py and not cmake.

Reply:
I can try using it but I'd much prefer use the default installation option. I actually tried using the python/setup.py before and it was not a nice experience.

Reply:
Maybe you also need to call https://github.com/equinor/segyio/blob/1.9.9/python/pyproject.toml#L24-L38 in preBuild. Not sure.

Reply:
I have no idea of how to do it. Nevertheless, do you insist on using buildPythonPackage? I'd be much happier otherwise.

Reply:
If the package should be part of pythonPackages than using buildPythonPackage will be a lot easier since we don't need to call several hooks by hand.

Reply:
Well, the package is working as is, and with the toPythonModule function it calls all those needed hooks afaik.

Reply:
Oh, missed that. Did you check if python39 and python310 work and don't mix versions?

Reply:
What do you mean by don't mixing versions? I will test those versions individually though and get back to you.

Reply:
If we build python310Packages.segyio it should not pull in python39 under any circumstances into this derivation.

Reply:
@SuperSandro2000 I tested that and from what I could gather there was not mixing. Still the build failed with python 3.10, hence I added a patch to fix the bug. It all builds  now.